config/config.go

```diff
@@ -153,22 +153,30 @@ func (c *Config) GetScopeStrategy() fosite.ScopeStrategy {
 }
 
 func matchesRange(r *http.Request, ranges []string) error {
-	ip, _, err := net.SplitHostPort(r.RemoteAddr)
+	forwardedFor := r.Header.Get("X-Forwarded-For")
```

Comment:
This has some issues, one of them is that `strings.Split` always returns at least one element (even if the string is empty) and that we shouldn't expect a whitespace after `,`. Let's rewrite this a bit:

```
remoteIP, _, err := net.SplitHostPort(r.RemoteAddr)
if err != nil {
	return errors.WithStack(err)
}

check := []string{remoteIP}
for _, fwd := stringsx.Split(r.Header.Get("X-Forwarded-For"), ",") {
    check = append(check, strings.Trim(fwd))
}
```